tests/stubtest_third_party.py

```diff
@@ -128,6 +137,80 @@ def run_stubtest(dist: Path, *, verbose: bool = False, specified_platforms_only:
     return True
 
 
+# Performs some black magic in order to run stubtest inside uWSGI
+# we have to write the exit code from stubtest to a surrogate file
+# because uwsgi --pyrun does not exit with the exitcode from the
+# python script. We have a second wrapper script that passed the
+# arguments along to the uWSGI script and retrieves the exit code
+# from the file, so it behaves like running stubtest normally would
+def setup_uwsgi_stubtest_command(dist: Path, venv_dir: Path, stubtest_cmd: list[str]) -> bool:
+    uwsgi_ini = dist / "@tests/uwsgi.ini"
+    if not uwsgi_ini.exists():
+        print_error("Did not find a uwsgi.ini for the uWSGI tests")
+        return False
+
+    uwsgi_script = venv_dir / "uwsgi_stubtest.py"
+    wrapper_script = venv_dir / "uwsgi_wrapper.py"
+    exit_code_surrogate = venv_dir / "exit_code"
+    uwsgi_script_contents = dedent(
+        f"""
+        import json
+        import os
+        import sys
+        from mypy.stubtest import main
+
+        sys.argv = json.loads(os.environ.get("STUBTEST_ARGS"))
+        exit_code = main()
+        with open("{exit_code_surrogate}", mode="w") as fp:
+            fp.write(str(exit_code))
+        sys.exit(exit_code)
+        """
+    )
+    uwsgi_script.write_text(uwsgi_script_contents)
+
+    if sys.platform == "win32":
+        uwsgi_exe = venv_dir / "Scripts" / "uwsgi.exe"
+    else:
+        uwsgi_exe = venv_dir / "bin" / "uwsgi"
+
+    if not uwsgi_exe.exists():
+        print_error("Did not find a uwsgi executable")
+        return False
+
+    # this wrapper script only forwards stdout, since stderr contains
+    # a bunch of spam from uWSGI, if you want to debug the internal script
```

Comment:
```suggestion
    # This wrapper script only forwards stdout, since stderr contains
    # a bunch of spam from uWSGI. If you want to debug the internal script,
```